src/ophyd_async/epics/core/_util.py

```diff
@@ -34,9 +36,12 @@ def get_supported_values(
     if issubclass(enum_cls, SubsetEnum):
         if not set(choices).issubset(pv_choices):
             raise TypeError(error_msg + "to be a subset of them.")
-    else:
+    elif issubclass(enum_cls, StrictEnum):
         if set(choices) != set(pv_choices):
             raise TypeError(error_msg + "to be strictly equal to them.")
+    elif issubclass(enum_cls, SupersetEnum):
+        if len(set(choices).intersection(pv_choices)) == 0:
+            raise TypeError(error_msg + ". None of them match.")
```

Comment:
This is checking that at least one of the returned values is in the superset enum type, is that the expected behaviour, or should all of the returned choices be in choices, i.e. the opposite of SubsetEnum        `set(pv_choices).issubset(choices):`? I'll read the ticket

Reply:
I'm not sure if I was clear in the ticket, but I meant what you said @DiamondJoseph